kpatch-build/kpatch-build

```diff
@@ -262,64 +262,64 @@ find_special_section_data() {
 		       e == 1 {printf("export EX_STRUCT_SIZE=%d\n", $4); e = 2}
 
 		       # Bail out once we have everything
-		       a == 2 && b == 2 && (p == 2 || skip_p) && e == 2 {exit}')
+		       a == 2 && b == 2 && (p == 2 || skip_p) && e == 2 {exit}')"
 
-	[[ -n $SPECIAL_VARS ]] && eval "$SPECIAL_VARS"
+	[[ -n "$SPECIAL_VARS" ]] && eval "$SPECIAL_VARS"
 
-	[[ -z $ALT_STRUCT_SIZE ]] && die "can't find special struct alt_instr size"
-	[[ -z $BUG_STRUCT_SIZE ]] && die "can't find special struct bug_entry size"
-	[[ -z $EX_STRUCT_SIZE ]]  && die "can't find special struct paravirt_patch_site size"
-	[[ -z $PARA_STRUCT_SIZE && $CONFIG_PARAVIRT -ne 0 ]] && die "can't find special struct paravirt_patch_site size"
+	[[ -z "$ALT_STRUCT_SIZE" ]] && die "can't find special struct alt_instr size"
+	[[ -z "$BUG_STRUCT_SIZE" ]] && die "can't find special struct bug_entry size"
+	[[ -z "$EX_STRUCT_SIZE" ]]  && die "can't find special struct paravirt_patch_site size"
+	[[ -z "$PARA_STRUCT_SIZE" && "$CONFIG_PARAVIRT" -ne 0 ]] && die "can't find special struct paravirt_patch_site size"
 
 	return
 }
 
 find_parent_obj() {
-	dir=$(dirname $1)
-	absdir=$(readlink -f $dir)
-	pwddir=$(readlink -f .)
-	pdir=${absdir#$pwddir/}
-	file=$(basename $1)
-	grepname=${1%.o}
-	grepname=$grepname\\\.o
-	if [[ $DEEP_FIND -eq 1 ]]; then
+	dir="$(dirname "$1")"
+	absdir="$(readlink -f "$dir")"
+	pwddir="$(readlink -f .)"
+	pdir="${absdir#$pwddir/}"
+	file="$(basename "$1")"
+	grepname="${1%.o}"
+	grepname="$grepname\.o"
+	if [[ "$DEEP_FIND" -eq 1 ]]; then
 		num=0
-		if [[ -n $last_deep_find ]]; then
-			parent=$(grep -l $grepname $last_deep_find/.*.cmd | grep -v $pdir/.${file}.cmd |head -n1)
-			num=$(grep -l $grepname $last_deep_find/.*.cmd | grep -v $pdir/.${file}.cmd |wc -l)
+		if [[ -n "$last_deep_find" ]]; then
+			parent="$(grep -l "$grepname" "$last_deep_find"/.*.cmd | grep -v "$pdir/.${file}.cmd" |head -n1)"
+			num="$(grep -l "$grepname" "$last_deep_find"/.*.cmd | grep -v "$pdir/.${file}.cmd" |wc -l)"
```

Comment:
Should `${file}` be quoted here (and below) in case it has spaces (probably won't ever happen)?

Reply:
I'm confused here as `${file}` is quoted by the surrounding quotes: `"$pdir/.${file}.cmd"`